static/js/src/advantage/users/app.tsx

```diff
@@ -1,16 +1,41 @@
 import React from "react";
 import ReactDOM from "react-dom";
+import { QueryClient, QueryClientProvider, useQuery } from "react-query";
 import AccountUsers from "./AccountUsers";
+import { requestAccountUsers } from "./api";
 
-// TODO: replace with real data once API integration is complete
-import { mockData } from "./mockData";
+const oneHour = 1000 * 60 * 60;
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      refetchOnWindowFocus: false,
+      refetchOnMount: false,
+      refetchOnReconnect: false,
+      staleTime: oneHour,
+      retryOnMount: false,
+    },
+  },
+});
 
-function App() {
-  return (
+const AccountUsersWithQuery = () => {
+  const { isSuccess, data } = useQuery("accountUsers", async () => {
+    const res = await requestAccountUsers();
+    return res;
+  });
+  return isSuccess && data ? (
     <AccountUsers
-      organisationName={mockData.organisationName}
-      users={mockData.users}
+      organisationName={data.organisationName}
+      users={data.users}
+      accountId={data.accountId}
     />
+  ) : null;
+};
+
+function App() {
+  return (
+    <QueryClientProvider client={queryClient}>
+      <AccountUsersWithQuery />
+    </QueryClientProvider>
```

Comment:
Here you re-render the whole app each time the `"accountUsers"` query updates. An alternative could be to only "subscribe" to this query the components that use it (`TableView` for example) thoughts?

Reply:
Yes you're right, I'll do some refactoring and optimisations in another PR. I have a task for "container component", it would fit right in.